apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -42,14 +100,36 @@ const aichatSlice = createSlice({
     setShowWarningModal: (state, action: PayloadAction<boolean>) => {
       state.showWarningModal = action.payload;
     },
+    updateChatMessageStatus: (
+      state,
+      action: PayloadAction<{id: number; status: Status}>
+    ) => {
+      const {id, status} = action.payload;
+      const chatMessage = state.chatMessages.find(msg => msg.id === id);
+      if (chatMessage) {
+        chatMessage.status = status;
+      }
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(submitChatMessage.fulfilled, state => {
+      state.isWaitingForChatResponse = false;
+    });
+    builder.addCase(submitChatMessage.rejected, (state, action) => {
+      state.isWaitingForChatResponse = false;
+      console.error(action.error);
```

Comment:
Should we go ahead and add an error field to Aichat state? We don't have to actually handle it in the view yet, but for now we could just do something like `state.chatMessageError = true` (doesn't necessarily have to be called `chatMessageError`)

Reply:
Sure! Just added.